src/mds/Server.cc

```diff
@@ -1118,6 +1118,14 @@ void Server::find_idle_sessions()
 	     << "), not marking any client stale" << dendl;
     return;
   }
+  // don't kick client if ops or osds are laggy
+  if(mds->objecter->get_num_osd_laggy() || mds->objecter->get_num_op_laggy()) {
+    mds->clog->warn() << "laggy OSDs deteced, skipping laggy clients eviction "
+                      << "until they return to normal";
+    dout(10) << "laggy OSDs deteced, skipping laggy clients eviction " 
+             << "untill they return to normal" << dendl;
+    return;
+  }
```

Comment:
Possibly you should do this in `evict_cap_revoke_non_responders()` instead ?

Reply:
possible, will check that code and update accordingly.

Reply:
If you do this here in `find_idle_sessions()` it will skip all the other non-cap-revoking cases, which should make no sense. As Patrick mentioned in the tracker the revoking could be blocked when the client is waiting for the OSD requests' responds.

In client side if there has dirty data and when flushing it will hold the `Fw` caps and only when OSD responses it will the client release this caps. So if the OSDs are laggy the responds maybe delayed.

Reply:
Then this commit makes no sense any more since you have fixed it in `evict_cap_revoke_non_responders()`.

Reply:
As patrick mentioned in https://github.com/ceph/ceph/pull/49971#discussion_r1121757643, it's good to check at both the places, `find_idle_sessions()` would catch straightforward laggy clients while clients not responding to release caps/forced delay in release due to laggy OSDs could be addressed in `evict_cap_revoke_non_responders()`.